vitrina/requests/views.py

```diff
@@ -44,3 +47,44 @@ def get_context_data(self, **kwargs):
         }
         context_data.update(extra_context_data)
         return context_data
+        
+        
+class RequestCreateView(CreateView):
+    model = Request
+    form_class = RequestForm
+    template_name = 'base_form.html'
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs['user'] = self.request.user
+        return kwargs
+
+    def get_context_data(self, **kwargs):
+        context_data = super().get_context_data(**kwargs)
+        context_data['current_title'] = _('Poreikio registravimas')
+        return context_data
+
+
+class RequestUpdateView(PermissionRequiredMixin, UpdateView):
+    model = Request
+    form_class = RequestForm
+    template_name = 'base_form.html'
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs['user'] = self.request.user
+        return kwargs
+
+    def has_permission(self):
+        try:
+            request = Request.objects.get(pk=self.kwargs['pk'])
+        except (ObjectDoesNotExist, KeyError):
+            request = None
+        if self.request.user.is_authenticated and request and request.user:
+            return self.request.user == request.user
+        return False
```

Comment:
Please leave `return super().has_permission()` if your permission check passes. We do also need to check other permissions too.